scripts/airports.ts

```
@@ -0,0 +1,165 @@
+  return airports
+    .filter(
+      (airport) =>
+        (airport.type === "large_airport" ||
+          airport.type === "medium_airport") &&
+        airport.scheduled_service === "yes",
+    )
+    .map((airport) => {
+      return {
+        iata: airport.iata,
+        country_code: airport.country_code,
+        country:
+          countriesTranslation.getName(airport.country_code, germanLocale) ??
+          "",
+        airport: translateAirportName(airport.airport),
+        latitude: airport.latitude,
+        longitude: airport.longitude,
+        city: translateAirportCity(airport),
+      };
+    });
+}
+
+function parseAiportDataSource(content: string): AiportDataSource[] {
+  const rows = content.split("\n");
+  const airports: AiportDataSource[] = [];
+  rows.slice(1).forEach((row) => {
+    const [
+      ,
+      ,
+      type,
+      name,
+      latitude,
+      longitude,
+      ,
+      ,
+      country_codeData,
+      ,
+      city,
+      scheduled_service,
+      ,
+      iataData,
+    ] = row.split(",");
+
+    const airportData = {
+      iata: iataData,
+      country_code: country_codeData,
+      airport: name,
+      latitude: latitude,
+      longitude: longitude,
+      city: city,
+      type: type,
+      scheduled_service: scheduled_service,
+    };
+
+    const airportResult = AirportDataSourceSchema.safeParse(airportData);
+
+    if (airportResult.success) {
+      airports.push(airportResult.data);
+    }
+  });
+
+  return airports;
+}
+
+async function fetchAllAirports(): Promise<Aiport[]> {
+  const res = await fetch(AIPORTS_URL_DATA_SOURCE, {
+    headers: {
+      "content-type": "text/csv;charset=UTF-8",
+    },
+  });
+
+  if (res.status === 200) {
+    const content = await res.text();
+    const airportsDataSource = parseAiportDataSource(content);
+    return filteredLargeMediumAirports(airportsDataSource);
+  }
+
+  console.log(`Error code ${res.status}. Status text: ${res.statusText}`);
+  return [];
```

Should we throw here instead of returning an empty array? 
I think failing fast in this case would make more sense, since if there is no data from the airports data source, no entries will be generated and written into the `data/airports/data.json` file. 
Otherwise, with only a log, if one is not paying attention to the logs (which may be the case), one may wonder why there is no data being written into the file.

---

I was thinking about this possibility, although if the file doesn't have any data, I would assume the person may see it once it commit the file `data/airports/data.json`.

But I'm fine to thrown an exception, if you prefer this approach instead of have a empty file data

---

Yes, let's throw :) I think it'd make it more clear.
Thank you!